core/src/main/java/org/infinispan/xsite/irac/DefaultIracManager.java

```diff
@@ -417,56 +419,72 @@ private CompletableSource sendUpdateBatch(Collection<? extends IracStateData> ba
          }
          return Completable.complete();
       }
-      IracPutManyCommand cmd = commandsFactory.buildIracPutManyCommand(size);
-      Collection<IracManagerKeyState> invalidState = new ArrayList<>(size);
-      Collection<IracManagerKeyState> validState = new ArrayList<>(size);
-      for (IracStateData data : batch) {
-         if (data.entry == null && data.tombstone == null) {
-            // there a concurrency issue where the entry and the tombstone do not exist (remove following by a put)
-            // the put will create a new state and the key will be sent to the remote sites
-            invalidState.add(data.state);
-            continue;
-         }
-         validState.add(data.state);
-         if (data.state.isExpiration()) {
-            cmd.addExpire(data.state.getKey(), data.tombstone);
-         } else if (data.entry == null) {
-            cmd.addRemove(data.state.getKey(), data.tombstone);
-         } else {
-            cmd.addUpdate(data.state.getKey(), data.entry.getValue(), data.entry.getMetadata(), data.entry.getInternalMetadata().iracMetadata());
+
+      AggregateCompletionStage<Void> aggregation = CompletionStages.aggregateCompletionStage();
+      for (IracXSiteBackup backup: asyncBackups) {
+         IracPutManyCommand cmd = commandsFactory.buildIracPutManyCommand(size);
```

Comment:
you can "fail" fast in 2 cases

* if `backup.isBackOffMode()` -> continue // there is nothing to do
* if backup is offline -> for each data in batch do `data.successFor(backup)` // otherwise, state is never removed.

Reply:
Updated for the `backup.isBackOffMode()`. For the offline, I kept getting some test failures.